crates/llm-chain-llama/src/batch.rs

```diff
@@ -0,0 +1,136 @@
+use llm_chain_llama_sys::{llama_batch, llama_batch_free, llama_seq_id};
+use std::ptr::null_mut;
+
+#[derive(Debug, Clone)]
+#[allow(dead_code)]
+pub struct LlamaBatch {
+    n_tokens: i32,
+    token: Vec<i32>,
+    embd: Vec<f32>,
+    pos: Vec<i32>,
+    n_seq_id: Vec<i32>,
+    seq_id: Vec<Vec<i32>>,
+    logits: Vec<bool>,
+    all_pos_0: i32,
+    all_pos_1: i32,
+    all_seq_id: i32,
+}
+
+impl LlamaBatch {
+    pub fn new_with_tokens(tokens: Vec<i32>, max_seq: i32) -> Self {
+        let pos = (0..tokens.len()).map(|p| p as i32).collect::<Vec<_>>();
+        let embd = vec![];
+        let n_seq_id = vec![max_seq; tokens.len()];
+        let seq_id = vec![vec![0; tokens.len()]; tokens.len()];
+        let logits = vec![false; tokens.len()];
+        let all_pos_0 = 0;
+        let all_pos_1 = 0;
+        let all_seq_id = 0;
+
+        Self {
+            n_tokens: tokens.len() as i32,
+            token: tokens,
+            embd,
+            pos,
+            n_seq_id,
+            seq_id,
+            logits,
+            all_pos_0,
+            all_pos_1,
+            all_seq_id,
+        }
+    }
+
+    pub fn set_token(&mut self, token: i32, pos: i32) {
+        self.n_tokens = 1;
+        self.token[0] = token;
+        self.pos[0] = pos;
+        self.n_seq_id[0] = 1;
+        self.seq_id[0][0] = 0;
+        self.logits[0] = true;
+    }
+
+    pub fn token_count(&self) -> usize {
+        self.n_tokens as usize
+    }
+
+    pub fn enable_logits(&mut self, pos: usize) {
+        self.logits[pos] = true;
+    }
+}
+
+impl Drop for LlamaBatch {
+    fn drop(&mut self) {
+        unsafe {
+            llama_batch_free(self.into());
+        }
+    }
+}
+
+impl From<llama_batch> for LlamaBatch {
```

Comment:
when do we use this?

Reply:
Actually I don't think it is used at all anymore. I'll remove it.